### DISCUSSION TITLE
Testing API: How to mark a "parent" test as complete once all children have completed?

### DISCUSSION THREAD

**Discussion by JustinGrote** [microsoft/vscode-discussions#672] [Extension Development QnA] [answered]
CC @connor4312 

this is a minor visual thing, but right now if I run a test suite, you see a lot of `queued` status on all the parents until the entire test run completes by calling `end`, at which point the parent updates to the aggregate status of its children.

What's the best way to make it so the parent reflects the children once all children have processed but before the test run has completed?

![image](https://github.com/microsoft/vscode-discussions/assets/15258962/f3ebce15-83d7-437c-bb26-766ce2626f3f)

**Comment by connor4312** [comment]
Are you using the latest version of VS Code? This sounds like the issue fixed in https://github.com/microsoft/vscode/pull/180746/commits/e2799e22d34f4b012107aa9d250e159a6f6a21ea

**Reply by JustinGrote** [reply]
I can reproduce in both stable and insiders. The way I read it, this isn't the same issue.
![image](https://github.com/microsoft/vscode-discussions/assets/15258962/096d55b1-c74a-45fc-85dd-f53bd6426850)

**Reply by connor4312** [reply, marked as answer]
Are you manually marking the parent items as queued? If so, I would not do that. Parent items do constant inherit the state of their children, but if they've been manually set to "queued", that's a higher priority state than "passed" state it might inherit from the children.

**Reply by JustinGrote** [reply]
Pretty sure this is it, I'll try only enqueuing testitems and not their parent contexts
https://github.com/pester/vscode-adapter/blob/4b18781f8db580e1c66b7895950c1d9bb83e0de8/src/pesterTestController.ts#L274-L277

**Reply by JustinGrote** [reply]
Yep, here's how I fixed it in my case.
https://github.com/pester/vscode-adapter/commit/5679ecd04bf17b01080db482199570d1d64eb822

I'd like to make a docs notes but I couldn't find where the .d.ts docs are in the repo, are they localized somewhere else? Thanks for the help @connor4312!

**Reply by connor4312** [reply]
.d.ts is here :) https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts API docs are generated from that.

Docs site docs are at https://github.com/microsoft/vscode-docs/blob/main/api/extension-guides/testing.md

**Reply by JustinGrote** [reply]
No idea why that didn't come up in github search, weird. Thanks.
